README.md

```diff
@@ -9,26 +9,54 @@
 
 ## Usage
 
+### Singular hook
+
+Recommended for [TypeScript](#typescript)
 ```js
-// instantiate hook API
-const hook = new Hook()
+// instantiate singular hook API
+const hook = new Hook.Singular()
 
 // Create a hook
 function getData (options) {
-  return hook('get', options, fetchFromDatabase)
+  return hook(options, fetchFromDatabase)
     .then(handleData)
     .catch(handleGetError)
 }
 
 // register before/error/after hooks.
 // The methods can be async or return a promise
-hook.before('get', beforeHook)
-hook.error('get', errorHook)
-hook.after('get', afterHook)
+hook.before(beforeHook)
+hook.error(errorHook)
+hook.after(afterHook)
 
 getData({id: 123})
 ```
 
+### Hook collection
+```js
+// instantiate hook collection API
+const hookCollection = new Hook.Collection()
+
+// Create a hook
+function getData (options) {
+  return hookCollection('get', options, fetchFromDatabase)
+    .then(handleData)
+    .catch(handleGetError)
+}
+
+// register before/error/after hooks.
+// The methods can be async or return a promise
+hookCollection.before('get', beforeHook)
+hookCollection.error('get', errorHook)
+hookCollection.after('get', afterHook)
+
+getData({id: 123})
+```
+
+### Behind the scenes
```

Comment:
Maybe rename the title to "`Hook.Singular` vs `Hook.Collection`"?

Reply:
Yup, that's better